client-app/ui-kit/composables/useQuantityValidationSchema.test.ts

```diff
@@ -0,0 +1,121 @@
+import { describe, expect, it } from "vitest";
+import { mockI18n } from "../mocks";
+import { useQuantityValidationSchema } from ".";
+
+describe("use-quantity-validation-schema", () => {
+  mockI18n();
+
+  it("quantity is positive", () => {
+    const { quantitySchema } = useQuantityValidationSchema({});
+
+    expect(quantitySchema.value.isValidSync(1)).toBeTruthy();
+    expect(quantitySchema.value.isValidSync(0)).toBeFalsy();
+    expect(quantitySchema.value.isValidSync(-1)).toBeFalsy();
+  });
+
+  it("quantity is integer", () => {
+    const { quantitySchema } = useQuantityValidationSchema({});
+
+    expect(quantitySchema.value.isValidSync(1)).toBeTruthy();
+    expect(quantitySchema.value.isValidSync(1.5)).toBeFalsy();
+  });
+
+  it("quantity is less than system limit", () => {
+    const { quantitySchema } = useQuantityValidationSchema({});
+
+    expect(quantitySchema.value.isValidSync(1)).toBeTruthy();
+    expect(quantitySchema.value.isValidSync(999999999)).toBeFalsy();
+  });
+
+  it("available quantity only", () => {
+    const { quantitySchema } = useQuantityValidationSchema({
+      availableQuantity: 5,
+    });
+
+    expect(quantitySchema.value.isValidSync(5)).toBeTruthy();
+    expect(quantitySchema.value.isValidSync(6)).toBeFalsy();
+  });
+
+  it("available quantity and correct mininimum quantity", () => {
+    const { quantitySchema } = useQuantityValidationSchema({
+      availableQuantity: 5,
+      minQuantity: 2,
+    });
+
+    expect(quantitySchema.value.isValidSync(3)).toBeTruthy();
+    expect(quantitySchema.value.isValidSync(1)).toBeFalsy();
+    expect(quantitySchema.value.isValidSync(6)).toBeFalsy();
+  });
+
+  it("available quantity and incorrect minimum quantity", () => {
+    const { quantitySchema } = useQuantityValidationSchema({
+      availableQuantity: 5,
+      minQuantity: 6,
+    });
+
+    expect(quantitySchema.value.isValidSync(4)).toBeTruthy();
+    expect(quantitySchema.value.isValidSync(6)).toBeFalsy();
```

Comment:
If available quantity is less then minimum quantity, system should always show error. So this test is incorrect.

Reply:
![image](https://github.com/VirtoCommerce/vc-theme-b2b-vue/assets/6369252/48757ffa-f4f4-49c4-9746-add78b536c71)